Tyk single dc (#50)
TykTechnologies/tyk-charts: tyk-single-dc/README.md

It includes:
- the Tyk Gateway, an open source Enterprise API Gateway (supporting REST, GraphQL, TCP and gRPC protocols)
- Tyk Pump, an analytics purger that moves the data generated by your Tyk nodes to any back-end. Furthermore, it has all the required modifications to easily connect to Tyk Cloud or Multi Data Center (MDCB) control plane.
- Tyk Dashboard, a license based features that allows you to interact with Tyk in a browser

## Introduction

By default, this chart installs following components as subcharts on a [Kubernetes](https://kubernetes.io/) cluster using the [Helm](https://helm.sh/) package manager.
| Component | Enabled by Default | Flag |
| --------- | ------------------ | ---- |
|Tyk Gateway |true  | n/a                    |
|Tyk Pump    |true | global.components.pump |
|Tyk Dashboard| true| global.components.dashboard | 

To enable or disable each component, change the corresponding enabled flag.

Also, you can set the version of each component through `image.tag`. You could find the list of version tags available from [Docker hub](https://hub.docker.com/u/tykio).

> ## Prerequisites

Reviewer: We also need mongo/postgres

* Kuberentes 1.19+
* Helm 3+
* Redis should already be installed or accessible by the gateway. For Redis installation instruction, please see https://tyk.io/docs/tyk-oss/ce-helm-chart/#recommended-via-bitnami-chart.

## Installing the Chart

To install the chart from Git repository in namespace `tyk` with the release name `tyk-single-dc`:

    helm repo add tyk-helm https://helm.tyk.io/public/helm/charts/
    helm repo update
    helm show values tyk-helm/tyk-single-dc > values-single-dc.yaml --devel

*If you use the bitnami chart for Redis installation, the DNS name of your Redis as set by Bitnami is `tyk-redis-master.tyk.svc.cluster.local:6379` You can update them in your local `values-single-dc.yaml` file under `global.redis.addr` and `global.redis.pass`. Alternatively, you can use `--set` flag to set it in Tyk installation. For example `--set global.redis.pass=$REDIS_PASSWORD`

*All the values above are just examples, please input the values specific for your deployment.

Then just run:

>     helm install tyk-single-dc tyk-single-dc -n tyk --create-namespace -f values-single-dc.yaml

Reviewer: ```suggestion
    helm install tyk-single-dc tyk-helm/tyk-single-dc -n tyk --create-namespace -f values-single-dc.yaml
```

## Uninstalling the Chart

```
helm uninstall tyk-single-dc -n tyk
```
This removes all the Kubernetes components associated with the chart and deletes the release.

## Upgrading Chart

```

> helm upgrade tyk-single-dc tyk-single-dc -n tyk

Reviewer: ```suggestion
helm upgrade tyk-single-dc tyk-helm/tyk-single-dc -n tyk
```

```

## Configuration

To get all configurable options with detailed comments:

>     helm show values tyk-single-dc > values-single-dc.yaml

Reviewer: ```suggestion
    helm show values tyk-helm/tyk-single-dc > values-single-dc.yaml
```

## Tyk Tyk single dc

> `tyk-single-dc` provides the default deployment of a Tyk Single Data Center. It will deploy the data center components with the settings provided in the values.yaml file.

Reviewer 2: ```suggestion
`tyk-single-dc` provides the default deployment of a Tyk Pro on single data center. It will deploy all required Tyk stack components with the settings provided in the values.yaml file.
```

It includes:
- the Tyk Gateway, an open source Enterprise API Gateway (supporting REST, GraphQL, TCP and gRPC protocols)
- Tyk Pump, an analytics purger that moves the data generated by your Tyk nodes to any back-end. Furthermore, it has all the required modifications to easily connect to Tyk Cloud or Multi Data Center (MDCB) control plane.

> - Tyk Dashboard, a license based component that allows you to interact with Tyk in a browser

Reviewer 2: ```suggestion
- Tyk Dashboard, a license based component that provides GUI management interface and analytics platform for Tyk
```

## Introduction

By default, this chart installs following components as subcharts on a [Kubernetes](https://kubernetes.io/) cluster using the [Helm](https://helm.sh/) package manager.
| Component | Enabled by Default | Flag |
| --------- | ------------------ | ---- |
|Tyk Gateway |true  | n/a                    |
|Tyk Pump    |true | global.components.pump |

> |Tyk Dashboard| true| global.components.dashboard | 

Reviewer 2: @andrei-tyk is tyk-dashboard an optional component? Can I turn it off to install additional gateway that connect to the dashboard?

Author: yes and yes :D

To enable or disable each component, change the corresponding enabled flag.

Also, you can set the version of each component through `image.tag`. You could find the list of version tags available from [Docker hub](https://hub.docker.com/u/tykio).

## Prerequisites

* Kuberentes 1.19+
* Helm 3+
* Redis should already be installed or accessible by the gateway. For Redis installation instruction, please see https://tyk.io/docs/tyk-oss/ce-helm-chart/#recommended-via-bitnami-chart.

## Installing the Chart

To install the chart from Git repository in namespace `tyk` with the release name `tyk-single-dc`:

    helm repo add tyk-helm https://helm.tyk.io/public/helm/charts/
    helm repo update

>     helm show values tyk-helm/tyk-single-dc > values-single-dc.yaml --devel

Reviewer 2: Please either use --devel for all commands or remove it..

Author: done

*If you use the bitnami chart for Redis installation, the DNS name of your Redis as set by Bitnami is `tyk-redis-master.tyk.svc.cluster.local:6379` You can update them in your local `values-single-dc.yaml` file under `global.redis.addr` and `global.redis.pass`. Alternatively, you can use `--set` flag to set it in Tyk installation. For example `--set global.redis.pass=$REDIS_PASSWORD`

*All the values above are just examples, please input the values specific for your deployment.

Then just run:

    helm install tyk-single-dc tyk-helm/tyk-single-dc -n tyk --create-namespace -f values-single-dc.yaml

## Uninstalling the Chart

```
helm uninstall tyk-single-dc -n tyk
```
This removes all the Kubernetes components associated with the chart and deletes the release.

## Upgrading Chart

```
helm upgrade tyk-single-dc tyk-helm/tyk-single-dc -n tyk
```

> ## Configuration

Reviewer 2: Please list all the Required configurations here, like dashboard license

Author: done
